File: app/schemas/vision.py
from pydantic import BaseModel, Field


class VisionSegmentResponse(BaseModel):
    has_tumor: bool
    segmented_image: str = Field(description="Overlay PNG encoded as base64.")
    mask: str = Field(description="Binary mask PNG encoded as base64.")
    model: str
    tumor_area_percentage: float
    average_tumor_probability: float
    latency_ms: float

File: deep-learning-frontend/src/types/vision.ts
export type BackboneType = 'resnet50' | 'efficientnet-b4'

export interface VisionResponse {
  has_tumor: boolean
  segmented_image: string
  mask: string
  model: string
  tumor_area_percentage: number
  average_tumor_probability: number
  latency_ms: number
}

export interface VisionError {
  status: 'error'
  code: string
  message: string
  detail: string | null
}
